code/Test_definitions/DeviceIdentifier_retrieve_identifier.feature

```diff
@@ -0,0 +1,141 @@
+@DeviceIdentifier_retrieve_identifier
+Feature: Camara Device Identifer API retrieve identifier
+
+# Input to be provided by the implementation to the tests
+# References to OAS spec schemas refer to schemas specified in code/API_definitions/CAMARA Mobile Device Identifier API.yaml
+# Implementation indications:
+# * api_root: API root of the server URL
+#
+# Testing assets:
+# * a mobile device with IMEI IMEI1
+# * a mobile device with IMEI IMEISV2, public IPv4 address PUBLICIPV4ADDRESS and public port PUBLICPORT
+# * a SIM card SIMCARD1 by telco1 and phone number PHONENUMBER1
+# * a SIM card SIMCARD2 by telco2 and phone number PHONENUMBER2
+
+  Background: Common Device Identifier retrieve identifer setup
+    Given the resource "/retrieve-identifier/v0"  as  base url
+    And the header "Content-Type" is set to "application/json"
+    And the header "Authorization" is set to a valid access token
+    And the header "x-correlator" is set to a UUID value
+    And the request body is compliant with the schema Device
+    
+  @DeviceIdentifier_retrieve_identifier0_phoneNumber_does_not_match_schema
+  Scenario Outline: phoneNumber value does not comply with the schema
+    Given the request body property "$.phoneNumber" is set to: <phone_number_value>
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+    Examples:
+      | phone_number_value |
+      | string_value       |
+      | 1234567890         |
+      | +12334foo22222     |
+      | +00012230304913849 |
+      | 123                |
+      | ++49565456787      |
+
+  @DeviceIdentifier_retrieve_identifier100_success
+  Scenario:  retrieve device identifer for phone number PHONENUMBER1, network connection and access token matches PHONENUMBER1
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the request body has the field phoneNumber with a value of PHONENUMBER1
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/200RetrieveIdentifier"
+    Then the response status code is 200
+    And the response property "$.imei" is IMEI1
```

Comment:
The test should check both that the correct IMEI is returned, and also that the response format is compliant with the defined schema for a 200 response.  Otherwise, the test test would not run at all if the "pre-condition" that the response format was correct was not satisfied, and the client cannot change the response format.
```suggestion
    Then the response status code is 200
    And the response body complies with the OAS schema at "/components/schemas/200RetrieveIdentifier"
    And the response header "x-correlator" has same value as the request header "x-correlator"
    And the response header "Content-Type" is "application/json"
    And the response property "$.imei" is IMEI1
```